core/src/main/java/tc/oc/pgm/goals/Goal.java

```diff
@@ -38,13 +38,10 @@
   boolean isShared();
 
   /**
-   * Returns true if the goal acts "normally". Normal behavior is defined when the goal is visible
-   * via mediums such as a {@link net.kyori.adventure.bossbar.BossBar}, the Scoreboard, and chat. If
-   * a call to this method returns false, this goal will not show up anywhere.
-   *
-   * <p>In most cases, this should simply delegate to {@link GoalDefinition#isVisible()}
+   * Returns true if the goal has the provided show flag enabled. Show flags define the goal
+   * behavior, making it visible only via certain mediums such as the scoreboard and chat.
    */
-  boolean isVisible();
+  boolean hasShowFlag(ShowOptions.ShowFlag flag);
```

Comment:
```suggestion
  boolean hasOption(ObjectiveOption option);
```

Let's use "option" so we don't confuse folks with actual flag objectives.